src/NHibernate/Cache/StandardQueryCache.cs

```diff
@@ -86,8 +85,15 @@ public void Clear()
 		{
 			// 6.0 TODO: inline the call.
 #pragma warning disable 612
-			return Put(key, returnTypes, result, queryParameters.NaturalKeyLookup, session);
+			var cached = Put(key, returnTypes, result, queryParameters.NaturalKeyLookup, session);
 #pragma warning restore 612
+
+			if (cached && key.ResultTransformer?.AutoDiscoverTypes == true && key.ResultTransformer.AutoDiscoveredAliases != null)
+			{
+				Cache.Put(new QueryAliasesKey(key), key.ResultTransformer.AutoDiscoveredAliases);
+			}
```

Comment:
In case we need aliases, this adds them to the cache for allowing retrieving them.

I though about using a local concurrent dictionary, but that would not work with distributed caches. We need theses aliases to be in distributed caches too.

Reply:
By the way, this works only for non-obsoleted `Get`. So, the bug would remain for users having an obsolete custom query cache implementation. I think this is acceptable.